tests/4.5/task/test_task_ThrdPrivate.c

```diff
@@ -0,0 +1,66 @@
+//===-- test_task_ThrdPrivate.c ------------------------------------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// Description
+// 1) The aim of the test case is to verify the functionality
+// of threadprivate construct when used under the influence of task construct
+// FuncABC() functions tests if thread 0 still has updated GlobalVar variable
+// while other threads observed GlobalVar still see 0
+// 2) In the main there is a check to verify GlobalVar is 0. This is because
+// what ever changes happen inside the task block are not reflected outside
+//===----------------------------------------------------------------------===//
+
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include "omp.h"
+#include "ompvv.h"
+#define THREADS 64
+
+#define INIT_VAL 9999
+
+
+int IfTstFailed = 0;
+
+int GlobalVar;
+#pragma omp threadprivate(GlobalVar)
+
+void FuncABC() {
+  int ThrdId = omp_get_thread_num();
+
+  {
+    if (ThrdId == 0) {
+      if ((GlobalVar != INIT_VAL)) {
+        IfTstFailed++;
+      }
+    } else {
+      if (GlobalVar != 0) {
+        IfTstFailed++;
+      }
+    }
+  }
+}
+
+
+int main() {
+  int errors = 0;
+  omp_set_num_threads(THREADS);
+  GlobalVar = INIT_VAL;
+#pragma omp parallel
+  {
+    int ThrdId = omp_get_thread_num();
+#pragma omp task if(0)
```

Comment:
What is purpose of if(0)? All threads will update GlobalVar

Reply:
Hi @andrewkallai ,
The purpose of if(0) is to generate an undefferred task so that the zeroth thread sees the updated GlobalVar value:
as per the openmp document: "When an if clause is present on a task construct, and the if clause expression evaluates to false, an undeferred task is generated, and the encountering thread must suspend the current task region, for which execution cannot be resumed until execution of the structured block that is associated with the generated task is completed. The use of a variable in an if clause expression of a task construct causes an implicit reference to the variable in all enclosing constructs."